tls/extensions/s2n_client_key_share.c

```diff
@@ -342,21 +339,35 @@ static int s2n_client_key_share_recv_pq_hybrid(struct s2n_connection *conn, stru
         return S2N_SUCCESS;
     }
 
+    /* The length of the hybrid key share must be one of two possible lengths. Its internal values are either length
+     * prefixed, or they are not. If actual_hybrid_share_size is not one of these two lengths, then
+     * s2n_is_tls13_hybrid_kem_length_prefixed() will return an error. */
+    uint16_t actual_hybrid_share_size = key_share->blob.size;
+    uint16_t unprefixed_hybrid_share_size = kem_group->curve->share_size + kem_group->kem->public_key_length;
+    uint16_t prefixed_hybrid_share_size = (2 * S2N_SIZE_OF_KEY_SHARE_SIZE) + unprefixed_hybrid_share_size;
+
     /* Ignore KEM groups with unexpected overall total share sizes */
-    if (key_share->blob.size != kem_group->client_share_size) {
```

Comment:
Looking at the old code, could you just do prefixed_hybrid_share_size=kem_group->client_share_size? Or, if we don't use client_share_size anywhere anymore, can we remove it? Removing it might be the clearer option, since that's no longer always the size of the share.

Reply:
Both client_share_size and server_share_size have been removed